src/python/pants/backend/jvm/targets/jvm_target.py

```diff
@@ -137,23 +137,33 @@ def strict_deps(self):
 
   @property
   def exports(self):
+    return self.payload.exports
+
+  @property
+  def exports_targets(self):
     """A list of exported targets, which will be accessible to dependents.
 
     :return: See constructor.
     :rtype: list
     """
-    exports = []
+    exports_targets = []
     for spec in self.payload.exports:
       addr = Address.parse(spec, relative_to=self.address.spec_path)
       target = self._build_graph.get_target(addr)
+      if target.is_thrift:
+        for dep in self.dependencies:
+          if dep != target and dep.is_synthetic and dep.derived_from == target:
```

Comment:
It's hard to tell what is going on here, because there isn't a comment. But it seems like this might reduce to checking `Target.concrete_derived_from`?

More generally, it seems like this code should move off of `Target`, and into the task. The direction that we're going to try and head in over time is that awareness of Task-specific dependencies (like `exports`), is a Task-level concern. So `jvm_compile` needs to know what `exports` and `strict_deps` mean, but no other Tasks need to.